sql/parser/ast.go

```diff
@@ -22,14 +22,14 @@ import (
 // consolidate a set of types, define the function as typeName().
 // This will help avoid name collisions.
 
-// Parse parses the sql and returns a Statement, which is the AST
-// representation of the query.
-func Parse(sql string) (Statement, error) {
+// Parse parses the sql and returns a Statement and the number of
+// variables. The Statement is the AST representation of the query.
+func Parse(sql string) (Statement, int, error) {
```

Comment:
This is fine for now, but I think the actual implementation should just walk the parse tree to determine a) the number of placeholders and b) to fill them in.